modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/config/DynamicConfigurationProvider.java

```diff
@@ -242,7 +243,7 @@ protected void registerConfiguration(ConfigurationInstance configuration) {
 
   private boolean isExpired(ConfigurationInstance configuration) {
     ConfigurationStats stats = configuration.getStatistics();
-    return stats.getRunningSources() == 0 && stats.getInflightOperations() == 0
+    return stats.getRunningSources() == 0 && stats.getInflightOperations() == 0 && stats.getActiveComponents() == 0
```

Comment:
same as before, it should be enough to just check for `stats.getActiveComponents()`